packages/common/src/services/auth/identity.ts

```diff
@@ -0,0 +1,114 @@
+import { SetAuthFn } from '@audius/hedgehog'
+import axios, { AxiosError, AxiosRequestConfig, AxiosResponse } from 'axios'
+import sigUtil from 'eth-sig-util'
+
+import { uuid } from '~/utils/uid'
+
+import { AuthHeaders } from './types'
+
+type Data = Record<string, unknown>
+type SetAuthFnParams = Parameters<SetAuthFn>[0]
+
+export type IdentityRequestError = AxiosError
+
+export type IdentityServiceConfig = {
+  identityServiceEndpoint: string
+}
+
+export class IdentityService {
+  identityServiceEndpoint: string
+
+  constructor({ identityServiceEndpoint }: IdentityServiceConfig) {
+    this.identityServiceEndpoint = identityServiceEndpoint
+  }
+
+  /* ------- HEDGEHOG AUTH ------- */
+
+  async getFn(params: {
+    lookupKey: string
+    username: string
+    visitorId?: string
+    otp?: string
+  }): Promise<{ iv: string; cipherText: string }> {
+    return await this._makeRequest({
+      url: '/authentication',
+      method: 'get',
+      params
+    })
+  }
+
+  async setAuthFn(obj: SetAuthFnParams) {
+    // get wallet from hedgehog and set as owner wallet
+    const ownerWallet = obj.wallet
+    // delete wallet object so it's not passed to identity
+    // @ts-ignore
+    delete obj.wallet
+
+    const unixTs = Math.round(new Date().getTime() / 1000) // current unix timestamp (sec)
+    const data = `Click sign to authenticate with identity service: ${unixTs}`
+    const signature = sigUtil.personalSign(ownerWallet.getPrivateKey(), {
+      data
+    })
+    const headers = {
+      [AuthHeaders.Message]: data,
+      [AuthHeaders.Signature]: signature
+    }
+
+    return await this._makeRequest({
+      url: '/authentication',
+      method: 'post',
+      headers,
+      data: obj
+    })
+  }
+
+  async setUserFn(obj: Data & { token?: string }) {
+    return await this._makeRequest({
+      url: '/user',
+      method: 'post',
+      data: obj
+    })
+  }
+
+  async getUserEvents(walletAddress: string) {
+    return await this._makeRequest<{ needsRecoveryEmail: boolean }>({
+      url: '/userEvents',
+      method: 'get',
+      params: { walletAddress }
+    })
+  }
+
+  /* ------- INTERNAL FUNCTIONS ------- */
+
+  async _makeRequest<T = unknown>(axiosRequestObj: AxiosRequestConfig) {
```

Comment:
I may follow up with this in the future and replace it with some kind of centralized request method? It makes me mildly uncomfortable that every object/class we have in client implements its own "make request" method, and most of them lean on axios while SDK uses native fetch.